packages/graph-retriever/tests/strategies/test_eager.py

```diff
@@ -79,11 +79,7 @@ async def test_animals_keywords(animals: Adapter, sync_or_async: SyncOrAsync):
         "alpaca",
         "bison",
         "cat",
-        "chicken",
```

Comment:
These tests now find fewer new things because of the adjacent K.

One interesting thing is that the `k` is "things with incoming edges from the outgoing edges". It does *not* mean `k new things`. So, after the first round or two, we may find the highest scoring "adjacent" edges keep cycling back to the same things.

On the one hand, this may mean we need to run with a higher `adjacent k`. But it also means that `adjacent_k` can then be used as a way to focus the search and not blow-up the result set as quickly.

Seems like this is OK and likely expected? But I think I hadn't fully thought through the fact that it would likely reduce the "new" things found.

Reply:
I was expecting that limiting to adjacent-k things instead of k-things per edge, would reduce the result set.